### PR TITLE
Depth parameter in policy graph generation

### PR DESCRIPTION
**Pull request by tanveergill**
##### Checklist

- [x] Tested in playground or other setup

### Summary by OpenAI

Release Notes:

- New Feature: Adds a new `depth` parameter to control the maximum depth of the graph when generating DOT and Mermaid files.
- New Feature: Adds a new `--graph-depth` parameter to the `aperturectl blueprints generate` command.
- New Feature: Adds a `--depth` parameter to the `aperturectl compile` command, allowing users to set the maximum depth for graph expansion.
- Documentation: Updates documentation for the above changes.

> "Graphs and circuits, oh my! 📈🔌
> With new parameters, we can now specify 
> The depth of our graphs, with ease and precision 
> Thanks to this PR, it's a cause for celebration!"

### CODE REVIEW

File: pkg/policies/controlplane/circuitfactory/graph.go [hunk 1]
```diff
@@ -26,6 +26,16 @@ func (circuit *Circuit) ToGraphView() *policymonitoringv1.CircuitView {
 	}
 }
 
+// TODO:
```
Comment (tanveergill) [inline pkg/policies/controlplane/circuitfactory/graph.go:29]: @openai,

Can you help implement this todo?

Comment (tanveergill) [reply pkg/policies/controlplane/circuitfactory/graph.go:29]: Need to make a few changes to the following code:
 1. Links should not be computed recursively by toGraphViewAux. Links within the children should not be part of links at the parent.
 2. When computing links at a tree certain links may reach up to the components in parent trees as well.
 3. Signals are uniquely identified by `Signal.SignalID()` which returns a `runtime.SignalID`:
```
type SignalID struct {
	SubCircuitID string
	SignalName   string
}
```

Comment (tanveergill) [reply pkg/policies/controlplane/circuitfactory/graph.go:29]: Can you please share the changes here? You do not have ability to make commits yet

Comment (harjotgill) [reply pkg/policies/controlplane/circuitfactory/graph.go:29]: Please generate go code based on above comment.

File: pkg/policies/controlplane/circuitfactory/graph.go
Comment (harjotgill) [inline pkg/policies/controlplane/circuitfactory/graph.go:1]: @openai visualize the logic in this file using ASCII based flowchart.

Comment (harjotgill) [reply pkg/policies/controlplane/circuitfactory/graph.go:1]: How can you visualize the logic and control flow in this file to help me understand it better? E.g. can you render ASCII, mermaid diagrams etc? Suggest 3 visualizations that you can generate to help me.

File: pkg/policies/controlplane/circuitfactory/tree.go [hunk 2]
```diff
@@ -0,0 +1,415 @@
+package circuitfactory
```
Comment (tanveergill) [inline pkg/policies/controlplane/circuitfactory/tree.go:1]: @openai,
Can you please review and summarize this file?

File: pkg/policies/controlplane/circuitfactory/auto-scale-components.go [hunk 3]
```diff
@@ -106,13 +106,14 @@ func newAutoScaleCompositeAndOptions(
...
 		if err != nil {
 			return retErr(err)
 		}
 
 		podScalerConfComp.PortMapping = portMapping
-		tree.Root = podScalerConfComp
+		tree.Node = podScalerConfComp
 
 		return tree, configuredComponents, fx.Options(options...), nil
 	} else if autoScaler := autoScaleComponentProto.GetAutoScaler(); autoScaler != nil {
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply pkg/policies/controlplane/circuitfactory/auto-scale-components.go:119]: This is a recursive structure, so node is fine

**Pull request merged**